pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -430,6 +431,12 @@ func (r *ReconcilePolicyRecommendation) Reconcile(ctx context.Context, request r
 		return reconcile.Result{}, nil
 	}
 
+	// Render a Policy Recommendation component for Windows if the cluster has Windows nodes.
```

Comment:
If I understand it correctly, we cannot have a cluster using policy recommendation and windows nodes at the same time, because windows does not support staged policies.

If that is the case, we should degrade the TigeraStatus immediately if hasWindowsNodes is true. I think we can move this if-statement further up, so we may save some unnecessary calculations, perhaps right underneath the ready checks at line 291.

Reply:
Moved the check to below line 291.

It isn't used, but I've kept the check in the render to enforce a defensive pattern